crane-cloud/backend
# Refactor/upgrades dependencies and config test

# Description
This PR includes several important updates and optimizations to the project.
- Key changes include, rewritten and optimized test cases, updated import statements based on upgraded packages and usage, a cleaned-up requirements.txt with updated dependencies, modified Alembic migration files and configurations, a refactored "SoftDeleteQuery" model file for better performance, and the removal of unused imports to enhance code readability and execution speed. These updates aim to maintain a clean, maintainable, and efficient codebase while improving the performance and reliability of the application.

The motivation behind these changes is to:
- Ensure the codebase remains clean, maintainable, and efficient.
- Improve the performance and reliability of the application.
- Align the project with the latest standards and best practices.
- Facilitate future development and integration efforts.

## Type of change

- [x] Chore (maintenance task such as refactoring, updating dependencies, improving documentation, or other non-functional changes)

## Trello Ticket ID
- NOT AVAILABLE

## How Can This Be Tested?
This can be tested by verifying the updates made in the files against the documentation for each used package and by following the steps required to set up and run the application as indicated in the readme.

- `@jwt_required` updated to `@jwt_required()`.
- `get_jwt_claims` updated to `get_jwt`.
- `user_claims_loader` updated to `additional_claims_loader`.
- Remove from `flask_sqlalchemy import Model` from `backend.app.models.transaction_records`.
- Rewrite `SoftDeleteQuery` class in the `bacend.app.models.model_mixin`.

## Screenshots
- Run Tests
![Screenshot from 2024-07-02 19-24-52](https://github.com/crane-cloud/backend/assets/50827537/312f24e5-7198-4da2-8d68-ffff10d2137f)

- Run Migrations
![Screenshot from 2024-07-02 19-40-45](https://github.com/crane-cloud/backend/assets/50827537/8f678b04-04d9-468b-a41e-9a0e400769f4)

- Run db upgrade
![Screenshot from 2024-07-02 19-41-20](https://github.com/crane-cloud/backend/assets/50827537/30cd8d28-b4b9-47fa-b7dc-690d92ef4d51)

- Database schema
![Screenshot from 2024-07-02 19-42-15](https://github.com/crane-cloud/backend/assets/50827537/8c0db443-3e13-4fcc-b936-5f994f550231)
